src/Kafka/Avro/Decode.hs

```diff
@@ -4,18 +4,19 @@ module Kafka.Avro.Decode
 , decodeWithSchema
 ) where
 
-import           Control.Monad.IO.Class (MonadIO)
-import           Data.Avro as A (FromAvro, Result(..), decode)
-import           Data.Avro.Schema (Schema)
-import           Data.Bits (shiftL)
+import           Control.Monad.IO.Class    (MonadIO)
+import           Data.Avro                 as A (FromAvro, Result (..), decode)
+import           Data.Avro.Schema          (Schema)
+import           Data.Bits                 (shiftL)
+import           Data.ByteString.Lazy      (ByteString)
+import qualified Data.ByteString.Lazy      as BL hiding (zipWith)
 import           Data.Int
-import           Data.ByteString.Lazy (ByteString)
-import qualified Data.ByteString.Lazy as BL hiding (zipWith)
 import           Kafka.Avro.SchemaRegistry
 
 data DecodeError = DecodeRegistryError SchemaRegistryError
                  | BadPayloadNoSchemaId
                  | DecodeError Schema String
+                 | InvalidSchema
```

Comment:
When schema content is invalid then `SchemaDecodeError` is returned:
https://github.com/haskell-works/hw-kafka-avro/blob/master/src/Kafka/Avro/SchemaRegistry.hs#L52

So `decode` would return `DecodeRegistryError SchemaDecodeError`